src/StochasticNoiseSimulator.cpp

```diff
@@ -5,31 +5,31 @@
 
 using CN = dd::ComplexNumbers;
 
-template<class DDPackage>
-std::map<std::string, std::size_t> StochasticNoiseSimulator<DDPackage>::Simulate(unsigned int shots) {
+template<class Config>
+std::map<std::string, std::size_t> StochasticNoiseSimulator<Config>::Simulate(std::size_t shots) {
     bool hasNonunitary = std::any_of(qc->cbegin(), qc->cend(), [&](const auto& p) { return p->isNonUnitaryOperation(); });
 
     if (!hasNonunitary) {
         perfectSimulationRun();
-        return Simulator<DDPackage>::MeasureAllNonCollapsing(shots);
+        return this->MeasureAllNonCollapsing(shots);
     }
 
     std::map<std::string, std::size_t> mCounter;
 
     for (unsigned int i = 0; i < shots; i++) {
         perfectSimulationRun();
-        mCounter[Simulator<DDPackage>::MeasureAll()]++;
+        mCounter[this->MeasureAll()]++;
     }
 
     return mCounter;
 }
 
-template<class DDPackage>
-void StochasticNoiseSimulator<DDPackage>::perfectSimulationRun() {
+template<class Config>
+void StochasticNoiseSimulator<Config>::perfectSimulationRun() {
     const auto nQubits = qc->getNqubits();
 
-    Simulator<DDPackage>::rootEdge = Simulator<DDPackage>::dd->makeZeroState(nQubits);
-    Simulator<DDPackage>::dd->incRef(Simulator<DDPackage>::rootEdge);
+    this->rootEdge = this->dd->makeZeroState(nQubits);
+    this->dd->incRef(this->rootEdge);
```

Comment:
posted a similar remark somewhere else, but could you remind me again, why the `this` is necessary here and the underlying object can't just simply be accessed without it?

Reply:
As I changed the template parameters, I must have messed up something as suddenly the lookup didn't work anymore (previously there was `Simulator<Config>::` instead of `this->`). I now changed it back, but tbh I don't really understand what was wrong :/